assets/user-profile/components/PauseNotificationModal.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {IUser} from 'interfaces';
+import {gettext} from 'utils';
+import {modalFormInvalid, modalFormValid} from 'actions';
+import {updateUserNotificationPause} from 'user-profile/actions';
+import Modal from 'components/Modal';
+
+interface IProps {
+    modalFormInvalid(): void;
+    modalFormValid(): void;
+    updateUserNotificationPause(schedule: Omit<IUser['notification_schedule'], 'last_run_time'>): void;
+    data: {
+        user: IUser;
+    };
+}
+
+interface IState {
+    pauseFrom?: string;
+    pauseTo?: string;
+}
+
+const disabledOptionDateFrom = new Date().toISOString().split('T')[0];
+
+class PauseNotificationModalComponent extends React.Component<IProps, IState> {
+    formRef: React.RefObject<HTMLFormElement>;
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            pauseFrom: this.props.data.user.notification_schedule?.pauseFrom ?? '',
+            pauseTo: this.props.data.user.notification_schedule?.pauseTo ?? '',
+        };
+
+        this.onSubmitForm = this.onSubmitForm.bind(this);
+        this.updateDate = this.updateDate.bind(this);
+        this.formRef = React.createRef<HTMLFormElement>();
+    }
+
+    componentDidMount() {
+        this.props.modalFormValid();
+    }
+
+    onSubmitForm(event: React.FormEvent<HTMLFormElement>) {
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (this.formRef.current == null) {
+            throw new Error('ref missing');
+        }
+
+        const inputs = this.formRef.current.querySelectorAll('input') as NodeListOf<HTMLInputElement>;
+
+        const hasErrors = [...inputs].some(input => {
+            return input.checkValidity() !== true;
+        });
+
+        if (hasErrors === true) {
+            this.formRef.current.reportValidity();
+        } else {
+            this.props.updateUserNotificationPause(this.state);
+        }
+    }
+
+    updateDate(event: any, state: 'pauseFrom' | 'pauseTo') {
+        const stateClone: IState = {};
+
+        stateClone[state] = event?.target.value;
+        this.setState(stateClone);
+    }
+
+    disabledMinOption(state: string | undefined) {
+        if (state != '' && state != undefined) {
+            const newMaxDate = new Date(state);
+            newMaxDate.setDate(newMaxDate.getDate() + 1);
+            return newMaxDate.toISOString().split('T')[0];
+        } else {
+            return disabledOptionDateFrom;
+        }
+    }
+
+    disabledMaxOption(state: string | undefined) {
+        if (state != '' && state != undefined) {
+            const newMaxDate = new Date(state);
+            newMaxDate.setDate(newMaxDate.getDate() - 1);
```

Comment:
same goes for here, I bet 1 is some kind of offset, I understand it but others would need to know also